components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -784,6 +784,10 @@ void AdsServiceImpl::ResetAllState(
       database_.release());
   VLOG_IF(1, !success) << "Failed to release database";
 
+  if (!connected()) {
+    return;
```

Comment:
you should still call `ResetState();` so that pref and files are deleted, even if process is not connected

Reply:
we should just do on line 776 `if (!should_shutdown || !connected()) {`

Reply:
Fixed